crates/bitcoin-da/src/verifier.rs

```diff
@@ -96,10 +96,13 @@ impl DaVerifier for BitcoinVerifier {
 
         // check that wtxid's of transactions in completeness proof are included in the InclusionMultiProof
         // and are in the same order as in the completeness proof
-        let mut iter = inclusion_proof.wtxids.iter();
-        completeness_proof
-            .iter()
-            .all(|tx| iter.any(|&y| y == tx.wtxid().to_byte_array()));
+        completeness_proof.iter().all(|tx| {
+            let wtxid = tx.wtxid();
+            inclusion_proof
+                .wtxids
+                .iter()
+                .any(|y| y == wtxid.as_byte_array())
```

Comment:
That's a good fix. This can be simplified with inclusion_proof.wtxids.contains(...).

And it shows that this part is O(N*M).

Reply:
Will look into using BTreeMap/BTreeSet's instead of looping over arrays in a separate PR, but it's a bit more tricky to test that. As we will need transactions in the blocks, it will probably require writing some sort of script/test.

But I suspect and it says in docs that HashSet's are inefficient compared to their efficiency it brings in normal architectures. But it's still worth to test imo.

Reply:
and now I actually suspect its a fix at all. because if you read the comment, it should also validate the order of items, in which makes it sensible to actually define iterator outside of loop to validate that the next element actually comes after.

but what I don't understand is, the output of `all` method is not used anywhere, there is no panic no errors. so even if it is false, we are not doing anything. anyone know why @eyusufatik @kpp @rakanalh @ercecan ?

Reply:
Not using all is a bug

Reply:
I don't see it boss. in outer iter, we expect everything to be true by using all. but in inner iter, we only need to find one equal value, so any.